### PR TITLE
Add swarm scheduler retry option

### PR DESCRIPTION
**Pull request by dongluochen**
Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

**Comment by vieux** [comment]
I'm not sure it's an issue but if you set the retry at 3, it's actually going to try 5 time

1) defaut code path
2) with the image affinity
3) retry1
4) retry2
5) retry3

otherwise LGTM thanks @dongluochen

**Comment by dongluochen** [comment]
Thanks @vieux. I update the swarm option to clarify createRetry. And take the imageNotFound retry into consideration. But always allow imageNotFound retry as before.

**Comment by vieux** [comment]
LGTM ping @jimmyxian @abronan @chanwit

File: cli/help.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ Arguments:
 Options:
    {{range .Flags}}{{.}}
    {{end}}{{if (eq .Name "manage")}}{{printf "\t * swarm.overcommit=0.05\tovercommit to apply on resources"}}
+                                    {{printf "\t * swarm.createretry=0\tcontainer create retry count after intial failure"}}
```
Comment (abronan) [inline cli/help.go]: `intial` -> `initial` :ghost:

**Comment by abronan** [comment]
thanks @dongluochen this one was much needed.

small typo on the command line but otherwise LGTM

**Comment by abronan** [comment]
@dongluochen Last thing :) can you please squash your commits? Thanks!

**Comment by dongluochen** [comment]
Squash done. Thanks @abronan.

**Comment by vieux** [comment]
:fireworks:

**Pull request merged**